build/pluginctl/logs_test.go

```diff
@@ -0,0 +1,202 @@
+		},
+		"one new entry, one old entry": {
+			logs:           []string{"old", "new"},
+			oldest:         "old",
+			expectedLogs:   []string{"new"},
+			expectedOldest: "new",
+			expectedAllNew: false,
+		},
+		"one new entry, multipile old entries": {
+			logs:           []string{"old1", "old2", "old3", "new"},
+			oldest:         "old3",
+			expectedLogs:   []string{"new"},
+			expectedOldest: "new",
+			expectedAllNew: false,
+		},
+		"multipile new entries, ultipile old entries": {
+			logs:           []string{"old1", "old2", "old3", "new1", "new2", "new3"},
+			oldest:         "old3",
+			expectedLogs:   []string{"new1", "new2", "new3"},
+			expectedOldest: "new3",
+			expectedAllNew: false,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			logs, oldest, allNew := checkOldestEntry(tc.logs, tc.oldest)
+
+			if allNew != tc.expectedAllNew {
+				t.Logf("expected allNew: %v, got %v", tc.expectedAllNew, allNew)
+				t.Fail()
+			}
+			if oldest != tc.expectedOldest {
+				t.Logf("expected oldest: %v, got %v", tc.expectedOldest, oldest)
+				t.Fail()
+			}
+
+			compareSlice(t, tc.expectedLogs, logs)
+		})
+	}
+}
+
+func TestFilterLogEntries(t *testing.T) {
+	now := time.Now()
+
+	for name, tc := range map[string]struct {
+		logs         []string
+		pluginID     string
+		since        time.Time
+		expectedLogs []string
+		expectedErr  bool
+	}{
+		"nil slice": {
+			logs:         nil,
+			expectedLogs: nil,
+			expectedErr:  false,
+		},
+		"empty slice": {
+			logs:         []string{},
+			expectedLogs: nil,
+			expectedErr:  false,
+		},
+		"no JSON": {
+			logs: []string{
+				`{"foo"`,
+			},
+			expectedLogs: nil,
+			expectedErr:  true,
+		},
+		"unknown time format": {
+			logs: []string{
+				`{"message":"foo", "plugin_id": "some.plugin.id", "timestamp": "2023-12-18 10:58:53"}`,
+			},
+			pluginID:     "some.plugin.id",
+			expectedLogs: nil,
+			expectedErr:  true,
+		},
+		"one matching entry": {
+			logs: []string{
+				`{"message":"foo", "plugin_id": "some.plugin.id", "timestamp": "2023-12-18 10:58:53.091 +01:00"}`,
+			},
+			pluginID: "some.plugin.id",
+			expectedLogs: []string{
+				`{"message":"foo", "plugin_id": "some.plugin.id", "timestamp": "2023-12-18 10:58:53.091 +01:00"}`,
+			},
+			expectedErr: false,
+		},
+		"filter out non plugin entries": {
+			logs: []string{
+				`{"message":"bar1", "timestamp": "2023-12-18 10:58:52.091 +01:00"}`,
+				`{"message":"foo", "plugin_id": "some.plugin.id", "timestamp": "2023-12-18 10:58:53.091 +01:00"}`,
+				`{"message":"bar2", "timestamp": "2023-12-18 10:58:54.091 +01:00"}`,
+			},
+			pluginID: "some.plugin.id",
+			expectedLogs: []string{
+				`{"message":"foo", "plugin_id": "some.plugin.id", "timestamp": "2023-12-18 10:58:53.091 +01:00"}`,
+			},
+			expectedErr: false,
+		},
+		"filter out old entries": {
+			logs: []string{
+				fmt.Sprintf(`{"message":"old2", "plugin_id": "some.plugin.id", "timestamp": "%s"}`, time.Now().Add(-2*time.Second).Format(timeStampFormat)),
```

Comment:
I think we should apply the fix at https://github.com/mattermost/mattermost-plugin-starter-template/pull/197/files while this PR is open